Obsidian/Services/HashtableService.cs

```diff
@@ -244,9 +244,13 @@ public string GetChunkPath(WadChunk chunk)
 
     public string GuessChunkPath(WadChunk chunk, WadFile wad)
     {
-        using Stream stream = wad.LoadChunkDecompressed(chunk).AsStream();
+        string extension = null;
+        if (chunk.Compression is not WadChunkCompression.Satellite)
+        {
+            using Stream stream = wad.LoadChunkDecompressed(chunk).AsStream();
 
-        string extension = LeagueFile.GetExtension(LeagueFile.GetFileType(stream));
+            extension = LeagueFile.GetExtension(LeagueFile.GetFileType(stream));
+        }
```

Comment:
What about this:
```cs
public static string GuessChunkPath(WadChunk chunk, WadFile wad)
{
    string extension = chunk.Compression switch
    {
        WadChunkCompression.Satellite => null,
        _ => GuessChunkExtension(chunk, wad)
    };

    return string.IsNullOrEmpty(extension) switch
    {
        true => string.Format("{0:x16}", chunk.PathHash),
        false => string.Format("{0:x16}.{1}", chunk.PathHash, extension),
    };

    static string GuessChunkExtension(WadChunk chunk, WadFile wad)
    {
        using Stream stream = wad.LoadChunkDecompressed(chunk).AsStream();

        return LeagueFile.GetExtension(LeagueFile.GetFileType(stream));
    }
}
```

Reply:
I don't really mind either way, so I've just applied this diff.